core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchResultLabelProvider.java

```diff
@@ -55,27 +64,28 @@ public StyledString getStyledText(Object element) {
     StyledString str = new StyledString();
 
     str.append(getLabelText(element));
-    str.append(" "); 
+    str.append(" ");
 
     if (element instanceof EObject) {
       boolean isLockedByOther = CapellaReadOnlyHelper.getReadOnlySectionHandler().isLockedByOthers((EObject) element);
       if (isLockedByOther) {
-        str.append("(Read-Only)", StyledString.DECORATIONS_STYLER); 
+        str.append("(Read-Only)", StyledString.DECORATIONS_STYLER);
       }
     }
     return str;
   }
 
   public StyledString getLabelText(Object element) {
-    StyledString labelText = new StyledString(""); 
+    StyledString labelText = new StyledString("");
     if (element instanceof CapellaSearchMatchEntry) {
       CapellaSearchMatchEntry capellaSearchMatchEntry = (CapellaSearchMatchEntry) element;
       EAttribute attribute = (EAttribute) capellaSearchMatchEntry.getAttribute();
       labelText.append(attribute.getName());
       labelText.append(": ");
       labelText.append(capellaSearchMatchEntry.getText());
-    }
-    else {
+    } else if (element instanceof Shape && ViewType.NOTE.equals(((Shape) element).getType())) {

```

Comment:
maybe we can use eclipse Adapters, to adapt something to something else.
That way, we can avoid hard dependencies everywhere, and being extensible quite easily.

https://github.com/eclipse/capella/blob/b05bc20ce8fc38b98ac7789c16c0f347d10d050b/tests/plugins/org.polarsys.capella.test.business.queries.ju/src/org/polarsys/capella/test/business/queries/ju/QueryAdapterFactory.java

https://github.com/eclipse/capella/blob/b05bc20ce8fc38b98ac7789c16c0f347d10d050b/tests/plugins/org.polarsys.capella.test.business.queries.ju/src/org/polarsys/capella/test/business/queries/ju/views/BqLabelProvider.java

Reply:
Hum I do not see how we can adapt these objects since they are totally different: A match entry (e.g. match on name), a diagram note and other Capella elements.